x-pack/plugins/enterprise_search/server/lib/indices/fetch_ml_inference_pipeline_processors.ts

```diff
@@ -79,7 +80,13 @@ export const fetchAndAddTrainedModelData = async (
     const trainedModelName = trainedModelData.model_id;
 
     if (pipelineProcessorData.hasOwnProperty(trainedModelName)) {
-      pipelineProcessorData[trainedModelName].modelType = trainedModelData.model_type || 'unknown';
+      const isBuiltIn = trainedModelData.tags.includes(BUILT_IN_MODEL_TAG);
+
+      pipelineProcessorData[trainedModelName].types = [
+        trainedModelData.model_type,
+        ...Object.keys(trainedModelData.inference_config),
```

Comment:
I _think_ we can rely on the `trainedModelData.model_type` and `trainedModelData.inference_config` being present, but I'm wondering if we should check to make sure (just to be certain)?

Reply:
What do you mean exactly that we should check to make sure?

Reply:
I think you'll just want to check that those values are not `undefined`

Reply:
Oh ya, that's what the `filter` two lines below this is doing 👍

Reply:
I guess I could do this:
```suggestion
        ...Object.keys(trainedModelData.inference_config || {}),
```